.github/workflows/ci.yml

```diff
@@ -80,3 +80,27 @@ jobs:
 
       - name: Run test
         run: ${{ matrix.test }}
+
+  push:
+
+    needs: test
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/master'
```

Comment:
Shame there's no simpler way.

Reply:
The event types would limit to pushes, but not to a particular branch.